contributors/design-proposals/admission_control_event_rate_limit.md

```diff
@@ -0,0 +1,149 @@
+# Admission control plugin: EventRateLimit
+
+## Background
+
+This document proposes a system for using admission control to enforce a limit
+on the number of event requests that the API Server will accept in a given time
+slice. In a large multi-tenant cluster, there may be a small percentage of
+tenants that are always in some type of error state, in which each tenant is
+producing a steady stream of error event requests. Each individual tenant may not
+be producing an absurd amount of event requests on its own, but taken collectively
+the errors from this small percentage of tenants can have a significant impact on
+the performance of the cluster overall. 
+
+## Use cases
+
+1. Ability to protect the API Server from being flooded by event requests.
+2. Ability to protect the API Server from being flooded by event requests for
+   a particular namespace.
+3. Ability to protect the API Server from being flooded by event requests for
+   a particular user.
```

Comment:
AFAIK user is not a kubernetes concept. cc @deads2k

Reply:
> AFAIK user is not a kubernetes concept. cc @deads2k

Users are a concept, post-authentication in the admission and REST handling chain, so I think this is correct.

Reply:
https://kubernetes.io/docs/admin/authentication/#users-in-kubernetes